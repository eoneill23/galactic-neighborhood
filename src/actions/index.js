export const addAPOD = APOD => ({
  type: 'ADD_APOD',
  APOD
});

export const addAPODError = error => ({
  type: 'APOD_ERROR',
  error
});

export const addISS = ISS => ({
  type: 'ADD_ISS',
  ISS
});

export const addISSError = error => ({
  type: 'ISS_ERROR',
  error
});

export const addAsteroids = asteroids => ({
  type: 'ADD_ASTEROIDS',
  asteroids
});

export const addAsteroidsError = error => ({
  type: 'ASTEROIDS_ERROR',
  error
});

export const addCalculation = calculation => ({
  type: 'ADD_CALCULATION',
  calculation
});

export const resetStoreCalculation = number => ({
  type: 'RESET_CALCULATION',
  number
});