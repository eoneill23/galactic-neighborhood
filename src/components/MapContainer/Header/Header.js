import React from 'react';
import { NavLink } from 'react-router-dom';
import './Header.css';

const Header = () => {
  return (
    <>
    <h1>Galactic Neighborhood</h1>
    <header className='header'>
      <NavLink activeClassName='active' className='nav-link' exact to='/'>Home</NavLink>
        <NavLink activeClassName='active' className='nav-link' to='/mercury'>Mercury</NavLink>
        <NavLink activeClassName='active' className='nav-link' to='/venus'>Venus</NavLink>
        <NavLink activeClassName='active' className='nav-link' to='/earth'>Earth</NavLink>
        <NavLink activeClassName='active' className='nav-link' to='/mars'>Mars</NavLink>
        <NavLink activeClassName='active' className='nav-link' to='/jupiter'>Jupiter</NavLink>
        <NavLink activeClassName='active' className='nav-link' to='/saturn'>Saturn</NavLink>
        <NavLink activeClassName='active' className='nav-link' to='/uranus'>Uranus</NavLink>
        <NavLink activeClassName='active' className='nav-link' to='/neptune'>Neptune</NavLink>
        <NavLink activeClassName='active' className='nav-link' to='/pluto'>Pluto</NavLink>
    </header>
    </>
  )
}

export default Header;