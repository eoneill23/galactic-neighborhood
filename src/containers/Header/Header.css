.header {
  background-color: red;
  height: 200px;
}

h1 {
  margin-top: 0;
}