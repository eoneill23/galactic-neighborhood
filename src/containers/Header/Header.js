import React from 'react';
import { NavLink } from 'react-router-dom';
import { resetStoreCalculation } from '../../actions/index';
import { connect } from 'react-redux';
import './Header.css';

export const Header = ({ resetStoreCalculation }) => {

  const dispatchNumber = 0;

  return (
    <>
    <h1>Galactic Neighborhood</h1>
    <header className='header'>
      <NavLink onClick={() => resetStoreCalculation(dispatchNumber)} activeClassName='active' className='nav-link' exact to='/'>Home</NavLink>
        <NavLink onClick={() => resetStoreCalculation(dispatchNumber)} activeClassName='active' className='nav-link' to='/mercury'>Mercury</NavLink>
        <NavLink onClick={() => resetStoreCalculation(dispatchNumber)} activeClassName='active' className='nav-link' to='/venus'>Venus</NavLink>
        <NavLink onClick={() => resetStoreCalculation(dispatchNumber)} activeClassName='active' className='nav-link' to='/earth'>Earth</NavLink>
        <NavLink onClick={() => resetStoreCalculation(dispatchNumber)} activeClassName='active' className='nav-link' to='/mars'>Mars</NavLink>
        <NavLink onClick={() => resetStoreCalculation(dispatchNumber)} activeClassName='active' className='nav-link' to='/jupiter'>Jupiter</NavLink>
        <NavLink onClick={() => resetStoreCalculation(dispatchNumber)} activeClassName='active' className='nav-link' to='/saturn'>Saturn</NavLink>
        <NavLink onClick={() => resetStoreCalculation(dispatchNumber)} activeClassName='active' className='nav-link' to='/uranus'>Uranus</NavLink>
        <NavLink onClick={() => resetStoreCalculation(dispatchNumber)} activeClassName='active' className='nav-link' to='/neptune'>Neptune</NavLink>
        <NavLink onClick={() => resetStoreCalculation(dispatchNumber)} activeClassName='active' className='nav-link' to='/pluto'>Pluto</NavLink>
    </header>
    </>
  )
}

export const mapDispatchToProps = dispatch => ({
  resetStoreCalculation: (dispatchNumber) => dispatch(resetStoreCalculation(dispatchNumber))
});

export default connect(null, mapDispatchToProps)(Header);