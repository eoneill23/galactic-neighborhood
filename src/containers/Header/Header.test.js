import React from 'react';
import { shallow } from 'enzyme';
import { resetStoreCalculation } from '../../actions/index';
import { Header, mapDispatchToProps } from './Header';

describe('Header', () => {
  let wrapper, mockResetStoreCalculation;

  beforeEach(() => {
    mockResetStoreCalculation = jest.fn();
    // mockDispatchNumber = 0;
    wrapper = shallow(<Header resetStoreCalculation={mockResetStoreCalculation}/>)
  })
  it('should match the snapshot with all data passed in correctly', () => {;

    expect(wrapper).toMatchSnapshot();
  });

  it('should call resetStoreCalculation when a NavLink is clicked', () => {

    wrapper.find('NavLink').at(0).simulate('click');

    expect(mockResetStoreCalculation).toHaveBeenCalledWith(0);
  });

  describe('mapDispatchToProps', () => {

    it('should call dispatch with 0 when resetStoreCalculation is called', () => {

      let mockDispatchNumber = 0;
      let mockDispatch = jest.fn();

      let actionToDispatch = resetStoreCalculation(mockDispatchNumber);
      let mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.resetStoreCalculation(mockDispatchNumber);

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });
});