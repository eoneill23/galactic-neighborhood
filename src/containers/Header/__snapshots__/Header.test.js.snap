// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match the snapshot with all data passed in correctly 1`] = `
<Fragment>
  <h1>
    Galactic Neighborhood
  </h1>
  <header
    className="header"
  >
    <NavLink
      activeClassName="active"
      className="nav-link"
      exact={true}
      onClick={[Function]}
      to="/"
    >
      Home
    </NavLink>
    <NavLink
      activeClassName="active"
      className="nav-link"
      onClick={[Function]}
      to="/mercury"
    >
      Mercury
    </NavLink>
    <NavLink
      activeClassName="active"
      className="nav-link"
      onClick={[Function]}
      to="/venus"
    >
      Venus
    </NavLink>
    <NavLink
      activeClassName="active"
      className="nav-link"
      onClick={[Function]}
      to="/earth"
    >
      Earth
    </NavLink>
    <NavLink
      activeClassName="active"
      className="nav-link"
      onClick={[Function]}
      to="/mars"
    >
      Mars
    </NavLink>
    <NavLink
      activeClassName="active"
      className="nav-link"
      onClick={[Function]}
      to="/jupiter"
    >
      Jupiter
    </NavLink>
    <NavLink
      activeClassName="active"
      className="nav-link"
      onClick={[Function]}
      to="/saturn"
    >
      Saturn
    </NavLink>
    <NavLink
      activeClassName="active"
      className="nav-link"
      onClick={[Function]}
      to="/uranus"
    >
      Uranus
    </NavLink>
    <NavLink
      activeClassName="active"
      className="nav-link"
      onClick={[Function]}
      to="/neptune"
    >
      Neptune
    </NavLink>
    <NavLink
      activeClassName="active"
      className="nav-link"
      onClick={[Function]}
      to="/pluto"
    >
      Pluto
    </NavLink>
  </header>
</Fragment>
`;
